src/Optimizer/ImageDimensions.php

```diff
@@ -162,6 +195,30 @@ public function hasHeight()
         return $this->getHeight() !== null;
     }
 
+    /**
+     * Check whether the width has a unit.
+     *
+     * @return bool Whether the width has a unit.
+     */
+    public function hasWidthUnit()
+    {
+        $width = $this->getWidth();
+
+        return is_string($width) && preg_match(self::UNIT_REGEX_PATTERN, $width);
+    }
+
+    /**
+     * Check whether the height has a unit.
+     *
+     * @return bool Whether the height has a unit.
+     */
+    public function hasHeightUnit()
+    {
+        $height = $this->getHeight();
+
+        return is_string($height) && preg_match(self::UNIT_REGEX_PATTERN, $height);
+    }
+
```

Comment:
Per above, I think these methods can be eliminated in favor of just checking of the get methods return a non-empty string.

Reply:
Removed via https://github.com/ampproject/amp-toolbox-php/pull/49/commits/b66999c3816b115d408c69fb9451b655093cbda8